packages/core/src/render3/error_code.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export const enum RuntimeErrorCode {
+  // Internal Errors
+
+  // Change Detection Errors
+  EXPRESSION_CHANGED_AFTER_CHECKED = '100',
+
+  // Dependency Injection Errors
+  INSTANTIATE_CYCLIC_DEPENDENCY = '200',
+  NO_PROVIDER = '201',
+  CONSTRUCTOR_NOT_COMPATIBLE = '202',
+
+  // Template Errors
+  MULTIPLE_COMPONENTS_MATCH = '300',
+  EXPORT_NOT_FOUND = '301',
+  PIPE_NOT_FOUND = '302',
+
+  // Styling Errors
+
+  // Declarations Errors
+  OUTPUT_NOT_INITIALIZED = '500',
+  UNEXPECTED_VALUE_IMPORTED_BY_MODULE = '501',
+
+  // i18n Errors
+
+  // Compilation Errors
+  TYPE_DOES_NOT_HAVE_EMOD_PROPERTY = '700'
```

Comment:
How about a more general name for this, not targeted at an implementation detail?

```suggestion
  TYPE_IS_NOT_AN_NGMODULE = '700'
```